pywbemtools/pywbemcli/_cmd_listener.py

```diff
@@ -0,0 +1,473 @@
+    `pywbemcli listener start` command which does exactly that.
+
+    There is no persisted data about the currently running listeners, instead
+    the currently running processes executing the `pywbemcli listener run`
+    command are the currently running listeners.
+
+    If such a listener process terminates, the corresponding listener no longer
+    exists, so there is no notion of a stopped listener nor does a listener
+    have an operational status. The `pywbemcli listener stop` command terminates
+    a listener gracefully.
+
+    In addition to the command-specific options shown in this help text, the
+    general options (see 'pywbemcli --help') can also be specified before the
+    'connection' keyword.
+    """
+    pass  # pylint: disable=unnecessary-pass
+
+
+@listener_group.command('run', cls=PywbemcliCommand,
+                        options_metavar=CMD_OPTS_TXT)
+@click.argument('name', type=str, metavar='NAME', required=True)
+@add_options(LISTEN_OPTIONS)
+@add_options(help_option)
+@click.pass_obj
+def listener_run(context, **options):
+    """
+    Run as a named WBEM indication listener.
+
+    Run this command as a named WBEM indication listener until it gets
+    terminated, e.g. by a keyboard interrupt, break signal (e.g. kill), or the
+    `pywbemcli listener stop` command.
+
+    A listener with that name must not be running, otherwise the command fails.
+
+    Examples:
+
+      pywbemcli listener run lis1
+    """
+    context.execute_cmd(lambda: cmd_listener_run(context, options))
+
+
+@listener_group.command('start', cls=PywbemcliCommand,
+                        options_metavar=CMD_OPTS_TXT)
+@click.argument('name', type=str, metavar='NAME', required=True)
+@add_options(LISTEN_OPTIONS)
+@add_options(help_option)
+@click.pass_obj
+def listener_start(context, **options):
+    """
+    Start a named WBEM indication listener in the background.
+
+    A listener with that name must not be running, otherwise the command fails.
+
+    Examples:
+
+      pywbemcli listener start lis1
+    """
+    context.execute_cmd(lambda: cmd_listener_start(context, options))
+
+
+@listener_group.command('stop', cls=PywbemcliCommand,
+                        options_metavar=CMD_OPTS_TXT)
+@click.argument('name', type=str, metavar='NAME', required=True)
+@add_options(help_option)
+@click.pass_obj
+def listener_stop(context, name):
+    """
+    Stop a named WBEM indication listener.
+
+    The listener is stopped gracefully.
+
+    A listener with that name must be running, otherwise the command fails.
+
+    Examples:
+
+      pywbemcli listener stop lis1
+    """
+    context.execute_cmd(lambda: cmd_listener_stop(context, name))
+
+
+@listener_group.command('show', cls=PywbemcliCommand,
+                        options_metavar=CMD_OPTS_TXT)
+@click.argument('name', type=str, metavar='NAME', required=True)
+@add_options(help_option)
+@click.pass_obj
+def listener_show(context, name):
```

Comment:
This could also show from a list of running listeners.

Reply:
You mean a user can interactively pick one? Is that really needed? How many listeners do you expect?

Reply:
Ignore until we better understand how people use listeners.